pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -354,51 +385,55 @@ func (n *NetworkPolicyController) createAppliedToGroup(npNsName string, pSel, nS
 	return appliedToGroupUID
 }
 
-// labelsMatchGroupSelector matches a Pod's labels to the
+// labelsMatchGroupSelector matches a ExternalEntity or Pod's labels to the
 // GroupSelector object and returns true, if and only if the labels
 // match any of the selector criteria present in the GroupSelector.
-func (n *NetworkPolicyController) labelsMatchGroupSelector(pod *v1.Pod, podNS *v1.Namespace, sel *antreatypes.GroupSelector) bool {
+func (n *NetworkPolicyController) labelsMatchGroupSelector(obj metav1.Object, ns *v1.Namespace, sel *antreatypes.GroupSelector) bool {
+	objSelector := sel.PodSelector
+	if _, ok := obj.(*v1alpha1.ExternalEntity); ok {
+		objSelector = sel.ExternalEntitySelector
+	}
 	if sel.Namespace != "" {
-		if sel.Namespace != pod.Namespace {
-			// Pods must be matched within the same Namespace.
+		if sel.Namespace != obj.GetNamespace() {
+			// Pods or ExternalEntities must be matched within the same Namespace.
 			return false
 		}
-		if !sel.PodSelector.Matches(labels.Set(pod.Labels)) {
-			// podSelector does not match the Pod's labels.
-			return false
+		if objSelector != nil && objSelector.Matches(labels.Set(obj.GetLabels())) {
```

Comment:
i believe the reason why we didnt check for objSelector for nil is because namespace is set, which means nsSelector is not present, which means the peer should at least have objSelector set

Reply:
This is needed now because when namespace is set, the objSelector can either be a podSelector or a externalEntitySelector, but not both. When this function is called with a Pod and the GroupSelector only has  externalEntitySelector, objSelector will be nil in that case. If we do not check nil here, there's going to be a panic

Reply:
ok thanks for the explanation..